common/app/controllers/EmailSignupController.scala

```diff
@@ -108,37 +128,74 @@ class EmailSignupController(
   def renderFormFromName(emailType: String, listName: String): Action[AnyContent] =
     csrfAddToken {
       Action { implicit request =>
-        val id = EmailNewsletter.fromIdentityName(listName).map(_.listIdV1)
-        id match {
-          case Some(listId) =>
-            Cached(1.day)(RevalidatableResult.Ok(views.html.emailFragment(emailLandingPage, emailType, listName)))
+        val identityNewsletter = EmailNewsletter.fromIdentityName(listName)
+        identityNewsletter match {
+          case Some(newsletter) =>
+            Cached(1.day)(
+              RevalidatableResult.Ok(
+                views.html.emailFragment(
+                  emailLandingPage,
+                  emailType,
+                  newsletter.identityName,
+                  newsletter.emailEmbed,
+                ),
+              ),
+            )
           case _ => Cached(15.minute)(WithoutRevalidationResult(NoContent))
         }
       }
     }
 
-  def subscriptionResult(result: String): Action[AnyContent] =
+  def subscriptionResult(result: String, listName: String): Action[AnyContent] =
     Action { implicit request =>
-      Cached(7.days)(result match {
-        case "success" => RevalidatableResult.Ok(views.html.emailSubscriptionResult(emailLandingPage, Subscribed))
-        case "invalid" => RevalidatableResult.Ok(views.html.emailSubscriptionResult(emailLandingPage, InvalidEmail))
-        case "error"   => RevalidatableResult.Ok(views.html.emailSubscriptionResult(emailLandingPage, OtherError))
-        case _         => WithoutRevalidationResult(NotFound)
-      })
-
+      listName match {
+        case "footer" => {
```

Comment:
`listName == "footer"` isn't what we should be checking is it?

Reply:
Got rid of this and separated out footer logic into its own method